drivers/gpio/gpio_shell.c

```diff
@@ -138,24 +187,36 @@ static int cmd_gpio_set(const struct shell *sh,
 static int cmd_gpio_blink(const struct shell *sh,
 			  size_t argc, char **argv)
 {
+	const struct gpio_driver_config *cfg;
 	const struct device *dev;
 	uint8_t index = 0U;
 	uint8_t value = 0U;
 	size_t count = 0;
 	char data;
+	int ret;
 
 	if (isdigit((unsigned char)argv[args_indx.index][0]) != 0) {
 		index = (uint8_t)atoi(argv[args_indx.index]);
 	} else {
-		shell_error(sh, "Wrong parameters for blink");
+		shell_error(sh, "Wrong parameters for GPIO blink");
 		return -EINVAL;
 	}
 	dev = device_get_binding(argv[args_indx.port]);
 
 	if (dev != NULL) {
-		index = (uint8_t)atoi(argv[2]);
-		shell_fprintf(sh, SHELL_NORMAL, "Blinking port %s index %d.", argv[1], index);
-		shell_fprintf(sh, SHELL_NORMAL, " Hit any key to exit");
+		cfg = (const struct gpio_driver_config *)dev->config;
+		if (cfg == NULL) {
+			shell_error(sh, "Device Configuration is NULL");
+			return -EIO;
+		}
+		if (index < find_msb_set(cfg->port_pin_mask)) {
+			shell_fprintf(sh, SHELL_NORMAL,
+			"Blinking port %s pin %u.", argv[1], index);
+			shell_fprintf(sh, SHELL_NORMAL, " Hit any key to exit");
+		} else {
+			shell_error(sh, "Invalid GPIO pin");
+			return -EINVAL;
+		}
```

Comment:
Same here - it really seems like the changes to the GPIO shell are a workaround for something in the new driver.

Reply:
same explanation for other comment